pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -26,27 +28,68 @@ type Check struct {
 	core.CheckBase
 	config         *checkconfig.CheckConfig
 	singleDeviceCk *devicecheck.DeviceCheck
+	discovery      discovery.Discovery
 }
 
 // Run executes the check
 func (c *Check) Run() error {
-
+	var checkErr error
 	sender, err := aggregator.GetSender(c.ID())
 	if err != nil {
 		return err
 	}
 
-	c.singleDeviceCk.SetSender(report.NewMetricSender(sender))
+	if c.config.IsDiscovery() {
+		var discoveredDevices []*devicecheck.DeviceCheck
+		discoveredDevices = c.discovery.GetDiscoveredDeviceConfigs()
+
+		jobs := make(chan *devicecheck.DeviceCheck, len(discoveredDevices))
+
+		var wg sync.WaitGroup
+
+		for w := 1; w <= c.config.Workers; w++ {
+			wg.Add(1)
+			go c.runCheckDeviceWorker(w, &wg, jobs)
+		}
+
+		for i := range discoveredDevices {
+			deviceCk := discoveredDevices[i]
+			deviceCk.SetSender(report.NewMetricSender(sender))
+			jobs <- deviceCk
```

Comment:
why do you pass devices to check through a worker, vs passing them as args ?

Reply:
We send all jobs to a `jobs` that is then consumed by any of X (configured) workers that is available.

> vs passing them as args ?

Do you have an example ? Not sure we can achieve the same this way.

Reply:
I see, I thought each worker had a dedicated `jobs` channel.